rita_client/src/operator_update/ops_websocket.rs

```diff
@@ -67,6 +73,34 @@ pub fn send_websocket_update() {
                                 // we only need to get the identity once
                                 let rita_client = settings::get_rita_client();
                                 let id = rita_client.get_identity().unwrap();
+                                let our_secretkey = match rita_client.network.wg_private_key {
+                                    Some(key) => SecretKey::from(key),
+                                    None => {
+                                        error!("No private key found, can't connect to operator server");
+                                        return;
+                                    }
+                                };
+                                let ops_pubkey;
+                                // we must receive the ops pubkey before we can proceed with encryption
+                                loop {
+                                    // check if we have received the ops pubkey
+                                    if let Ok(Some(msg)) =
+                                        timeout(SOCKET_CHECKER_TIMEOUT, ws.next()).await
+                                    {
+                                        let msg = msg.unwrap();
+                                        if let Some(hour) = handle_received_operator_message(msg, &our_secretkey, None) {
+                                            match hour {
+                                                ReceivedOpsData::WgKey(public_key) => {
```

Comment:
this is a good structure, only two comments

1) this key should also be used over in the heartbeat code, lazy static or setting it / getting it from the config would be appropriate
2) this should happen any time we get the message not just startup, just in case odd things happen. I do see the appeal of just crashing and trying again but I'm worried if we have some odd ordering bug we'll have a crash loop here go on forever.

Reply:
HEARTBEAT_SERVER_KEY now comes from ops over websocket- so if the router cannot connect via socket first it cannot use the dummy key to heartbeat